CONTRIBUTING.md

```diff
@@ -0,0 +1,105 @@
+not *more* likely to be mapped soon.
+
+Of course, keep in mind that some names are derived from official terminology - those are likely to be identical across all
+projects. If the code's strings imply something is a "VoxelShape", say, then you don't need to force yourself to come up with
+a different name. In addition, if you can show your thinking process to be independent - say, if something is obviously a
+"BlockPredicate" - that's probably fine as well.
+
+In general, this rule is somewhat vague, as a lot of the surrounding issues are somewhat vague. As such, we can only clearly
+define extreme situations and try to give a general outline of what we believe should and should not be done. If in doubt, ask
+more experienced contributors. (Of course, if you have never worked with decompiled Minecraft code previously, little of this is
+likely to apply to you.)
+
+### 3. Stick to common coding conventions.
+
+Mappings should, of course, stick to the [standard Java naming conventions](https://www.oracle.com/technetwork/java/javase/documentation/codeconventions-135099.html#367).
+Packages should be all lower case, class/enum/interface names should be `TitleCase`, fields and methods should be `camelCase`,
+and constants should be `UPPER_CASE`.
+
+There are additional conventions we have generally agreed to. Some are listed below - when in doubt, follow what the mappings do!
+
+* For getters and setters, we always use the "get"/"set" prefixes, with the exception of booleans - where verbs such as
+  "is", "should", "can", "does", etc. are used depending on what flows most naturally.
+
+### 4. Use a natural (English) word order.
+
+As a group, we've decided to focus on a more natural word order for our mappings. The most notable example of this in our
+mappings is that we use a *suffix* system instead of a *prefix* system. That would mean using something like `LoomBlock` and
+`LoomBlockEntity` instead of `BlockLoom` and `BlockEntityLoom`. This should be applied to methods and fields as well as
+classes.
+
+In keeping with the natural English feel, we prefer to avoid prefixing interfaces and enums with `I` and `Enum` respectively.
+While they can be useful for differentiation, we feel that in most cases there should be enough room to differentiate
+classes without requiring prefixes. In short: the ideal is for field accesses and method calls to feel like sentences, making
+it easier to reason about code.
+
+### 5. Make names explicit and efficient.
+
+When someone calls a method, they should know exactly what it will do and exactly what will result from it. While this can be
+found out by looking at the method code, it's a much better experience for a dev to be able to find out by seeing the name.
+Instead of using terms like `serialize` and `deserialize`, use something like `toJson` and `fromJson` or `toNbt` and
+`fromNbt`, depending on what the result of the method is. Methods should only have matching names if they do the same thing
```

Comment:
"deserialize" has a different meaning than "fromNbt" and "readNbt". The first implies the object being read is not the class itself and the class with the method is just a serializer for that object. The second implies it's a static method creating instances of the owner class from NBT. The third implies it's a non static method loading data from NBT into the already created instance.

So we shouldn't say we should always avoid "serialize" and "deserialize".

Also, are you saying that we should always add the paraneter type (json/nbt in this case) to the method's name rather than overloading methods? That's what it sounds like from what I read.

Reply:
We have decided to, so far, always avoid serialize and deserialize, even with serializer classes.

Reply:
>The second implies it's a static method creating instances of the owner class from NBT.

Except we're using it for non-static methods right now in about half the codebase... :thinking: